### PR TITLE
Add Raspberry Pi3 Uart

### PR DESCRIPTION
**Pull request by miyazakh**
Add use case on Raspberry Pi 3 B+ HW

### CODE REVIEW

File: hal/raspi3.c [hunk 1]
```diff
@@ -18,23 +18,209 @@
...
+char uart_read(void)
+{
+    char c;
+    /* wait until data is comming */
+    do{
+        asm volatile("nop");
+    } while(*UART0_FR & 0x10);
+    /* read it and return */
+    c = (char)(*UART0_DR);
+    /* convert carrige return to newline */
```
Comment (danielinux) [inline hal/raspi3.c:155]: spelling: carriage. Is this conversion really required?

File: hal/raspi3.c [hunk 2]
```diff
@@ -18,23 +18,209 @@
...
+}
+
+/**
+ * Send string to UART
+ */
+void uart_write(const char* buf, uint32_t sz) {
+    uint32_t len = sz;
+
+    while (len > 0 && *buf) {
+        /* convert newline to carrige return + newline */
```
Comment (danielinux) [inline hal/raspi3.c:169]: spelling, plus conversion from CR to CR+NL, that should be handled by the caller.

**Review by danielinux — CHANGES_REQUESTED** [review]
Good addition. Thanks! Just left a couple of comments on the spelling of "carriage" in "carriage return". In the same context, I'm wondering if these uart API should not handle the CR to CR+LF conversion at all.

File: hal/raspi3.c (hunk 2 above)
Comment (miyazakh) [reply hal/raspi3.c]: Thanks. Agree. Carriage return code should be handled by the caller.

**Review by danielinux — APPROVED** [review]

**Pull request merged**